patches/tModLoader/Terraria/ModLoader/Default/ModAccessorySlotPlayer.cs

```diff
@@ -193,16 +285,242 @@ public override void SendClientChanges(ModPlayer clientPlayer)
+		}
+
+		public int LoadoutIndex { get; }
+
+		public EquipmentLoadout LoadoutReference { get; }
+
+		/// <inheritdoc cref="ModAccessorySlotPlayer.exAccessorySlot"/>
+		public Item[] ExAccessorySlot { get; private set; } = [];
+
+		/// <inheritdoc cref="ModAccessorySlotPlayer.exDyesAccessory"/>
+		public Item[] ExDyesAccessory { get; private set; } = [];
+
+		/// <inheritdoc cref="ModAccessorySlotPlayer.exHideAccessory"/>
+		public bool[] ExHideAccessory { get; private set; } = [];
+
+		public void SaveData(TagCompound tag)
+		{
+			tag[identifier] = new TagCompound {
+				["items"] = ExAccessorySlot.Select(ItemIO.Save).ToList(),
+				["dyes"] = ExDyesAccessory.Select(ItemIO.Save).ToList(),
+				["hidden"] = ExHideAccessory.ToList(),
+			};
+		}
+
+		/// <summary>
+		/// Loads data for this loadout and updates this instance accordingly.
+		/// Returns a collection of <see cref="SlotInfo"/> objects for slots, which are not added to the loadout,
+		/// because <see cref="ModAccessorySlot.HasEquipmentLoadoutSupport"/> changed since the last save.
+		/// </summary>
+		/// <param name="tag">The <see cref="TagCompound"/> from which to load the data</param>
+		/// <param name="order">Saved slot names in order.</param>
+		/// <param name="slots">Slot name to slot info mapping.</param>
+		/// <returns>
+		/// A collection of <see cref="SlotInfo"/> objects for slots, which are not added to the loadout,
+		/// because <see cref="ModAccessorySlot.HasEquipmentLoadoutSupport"/> changed since the last save.
+		/// </returns>
+		public IReadOnlyList<SlotInfo> LoadData(
+			TagCompound tag,
+			List<string> order,
+			Dictionary<string, (int SlotType, bool HasLoadoutSupport)> slots)
+		{
+			List<SlotInfo> result = [];
+			IList<Item> items;
+			IList<Item> dyes;
+			IList<bool> visible;
+
+			this.ResetAndSizeAccessoryArrays(slots.Count);
+
+			tag = tag.GetCompound(identifier);
+			items = tag.GetList<TagCompound>("items").Select(ItemIO.Load).ToList();
+			dyes = tag.GetList<TagCompound>("dyes").Select(ItemIO.Load).ToList();
+			visible = tag.GetList<bool>("hidden").ToList();
+
+			for (int i = 0; i < order.Count; i++) {
+				(int type, bool hasLoadoutSupport) = slots[order[i]];
+
+				Item dye = dyes.ElementAtOrDefault(i) ?? new Item();
+				Item accessory = items.ElementAtOrDefault(i) ?? new Item();
+				Item vanityItem = items.ElementAtOrDefault(i + order.Count) ?? new Item();
+				bool isHidden = visible.ElementAtOrDefault(i);
+
+				// If this slot has any acc/van/dye item, but doesn't have loadout support, the items shouldn't be in the loadout at all.
+				// We can try to move them to the inventory or try to put them in empty acc slots, but it might get complicated with slots and restrictions. Also the user might not want it there. They will be returned to the player OnEnterWorld.
+				bool hasLoadoutSupportSettingForSlotChanged = !hasLoadoutSupport && (!dye.IsAir || !accessory.IsAir || !vanityItem.IsAir);
```

Comment:
Can simplify this with above ^. Just `if (!hasLoadoutSupport)` add each non-air item to `itemsToDrop`